test/support/slack_client.ex

```diff
@@ -173,49 +130,39 @@ defmodule Cog.Test.Support.SlackClient do
     {:noreply, state}
   end
   def handle_info({{ref, sender}=caller, {:chat_and_wait, room, message, reply_from, nil}}, state) do
-    :timer.sleep(@api_wait_interval)
+    api_wait()
     result = Slack.Web.Chat.post_message(room, %{as_user: true, text: message, token: state.token, parse: :full})
     case result["ok"] do
       false ->
         send(sender, {ref, {:error, result["error"]}})
         {:noreply, state}
       true ->
-        user_id = find_user!(reply_from, state)
+        user_id = find_user_id!(reply_from, state)
         SlackClientState.add_waiter(user_id, caller)
         {:noreply, state}
     end
   end
   def handle_info({{ref, sender}=caller, {:chat_and_wait, room, message, reply_from, edited}}, state) do
-    :timer.sleep(@api_wait_interval)
+    api_wait()
     result = Slack.Web.Chat.post_message(room, %{as_user: true, text: message, token: state.token})
     case result["ok"] do
       false ->
         send(sender, {ref, {:error, result["error"]}})
         {:noreply, state}
       true ->
+        api_wait()
```

Comment:
I wonder if it'd be cleaner to have a version of `api_wait` that took either a fun or an MFA and waited before executing the function and returned the results like so:

```
result = api_wait({Slack.Web.Chat, :update, [result["channel"], edited, result["ts"], 
                                             %{token: state.token, as_user: true, parse: :full}])
```